### ISSUE TITLE
some issues bout grads

### ISSUE THREAD

**Issue by sjlljs176** [HealthX-Lab/MedCLIP-SAM#2]
I noticed that the grads_list was nan when using the BiomedCLIP. Is it normal? I feel weired.

This is the line.
https://github.com/HealthX-Lab/MedCLIP-SAM/blob/38e4cadfd607e32d8144427fc9d6f5cc72f64c09/saliency_maps/pytorch_grad_cam/base_cam.py#L192C13-L193C75

**Comment by TahaKoleilat** [comment]
Hi @sjlljs176,

Can you please provide snapshots of the output you are getting? I am running the code on my machine and I am not getting any NaN values in the grads_list. This could be maybe due to the input. So, if you can also please provide sample inputs where you are getting NaN values, it will be easier to reproduce and pinpoint the problem.

Thank you.

**Comment by sjlljs176** [comment]
the values in activations_and_grads.gradients hava the shape like (1,197,768) 
but the reshape_transfrom make it to be a list with nothing
![image](https://github.com/user-attachments/assets/74442b7b-df71-45df-982b-fed727e38c97)

**Comment by sjlljs176** [comment]
when debug in the get_cam_weights, the "importance" become a tensor with NaN values. I used the images of BUSI dataset as inputs.
![image](https://github.com/user-attachments/assets/c2bfc236-2a33-4812-b43e-385808c06c72)

**Comment by TahaKoleilat** [comment]
The issue was with the gradients shape. I added the fix [here](https://github.com/HealthX-Lab/MedCLIP-SAM/blob/main/saliency_maps/pytorch_grad_cam/base_cam.py#L186). 

Thank you for pointing this out @sjlljs176.

**Comment by sjlljs176** [comment]
Thanks for your help and answers.

**Comment by sjlljs176** [comment]
Thanks a lot  for your time and assistance!

**Comment by sjlljs176** [comment]
b.t.w, I found the value of loss became < 0, when i fine-tuned the BiomedCLIP with MedPix dataset. Is it normal?

**Comment by TahaKoleilat** [comment]
It's totally normal. Typical contrastive losses will be bounded between 0 and 1, however when the positive and negative pairs are decoupled the loss will be unbounded and thus can become negative.

**Comment by sjlljs176** [comment]
thanks a lot.
